source/material_model/rheology/dislocation_creep.cc

```diff
@@ -37,54 +37,93 @@ namespace aspect
-                                                      (constants::gas_constant*temperature*stress_exponents_dislocation_composition)) *
-                                             std::pow(strain_rate,((1. - stress_exponents_dislocation_composition)/stress_exponents_dislocation_composition));
+        const double viscosity_dislocation = 0.5 * std::pow(p.prefactor,-1/p.stress_exponent) *
+                                             std::exp((p.activation_energy + pressure*p.activation_volume)/
+                                                      (constants::gas_constant*temperature*p.stress_exponent)) *
+                                             std::pow(strain_rate,((1. - p.stress_exponent)/p.stress_exponent));
         return viscosity_dislocation;
       }
 
 
+      template <int dim>
+      std::pair<double, double>
+      DislocationCreep<dim>::compute_strain_rate_and_derivative (const double stress,
+                                                                 const double pressure,
+                                                                 const double temperature,
+                                                                 const DislocationCreepParameters creep_parameters) const
+      {
+        // Power law creep equation:
+        //   edot_ii_partial = A * stress^n * exp(-(E + P*V)/(RT))
+        //   d(edot_ii_partial)/d(stress) = A * n * stress^(n-1) * exp(-(E + P*V)/(RT))
+        // A: prefactor, edot_ii_partial: square root of second invariant of deviatoric strain rate tensor attributable to the creep mechanism,
+        // stress: deviatoric stress, E: activation energy, P: pressure,
+        // V; activation volume, n: stress exponent, R: gas constant, T: temperature.
+        const double strain_rate_dislocation = creep_parameters.prefactor *
+                                               std::pow(stress,creep_parameters.stress_exponent) *
+                                               std::exp(-(creep_parameters.activation_energy + pressure*creep_parameters.activation_volume)/
+                                                        (constants::gas_constant*temperature));
+
+        const double dstrain_rate_dstress_dislocation = creep_parameters.prefactor *
+                                                        creep_parameters.stress_exponent *
+                                                        std::pow(stress,creep_parameters.stress_exponent-1.) *
+                                                        std::exp(-(creep_parameters.activation_energy + pressure*creep_parameters.activation_volume)/
+                                                                 (constants::gas_constant*temperature));
+
+        const std::pair<double, double> strain_rate_and_derivative (strain_rate_dislocation, dstrain_rate_dstress_dislocation);
+        return strain_rate_and_derivative;
```

Comment:
same simplification as in diffusion_creep

Reply:
Yup. Done